? ./src/tiff/tiff.dsp_old
? ./src/tiff/tiff.dsw_old
Index: ./src/tiff/tif_compress.c
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/tiff/tif_compress.c,v
retrieving revision 1.2
diff -b -u -2 -r1.2 tif_compress.c
--- ./src/tiff/tif_compress.c	2000/02/16 06:26:12	1.2
+++ ./src/tiff/tif_compress.c	2002/10/20 04:15:25
@@ -146,5 +146,5 @@
 
 int
-TIFFSetCompressionScheme(TIFF* tif, int scheme)
+TIFFSetCompressionScheme(TIFF* tif, uint16 scheme)
 {
 	const TIFFCodec *c = TIFFFindCODEC(scheme);
Index: ./src/tiff/tif_dir.c
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/tiff/tif_dir.c,v
retrieving revision 1.1
diff -b -u -2 -r1.1 tif_dir.c
--- ./src/tiff/tif_dir.c	1999/11/30 18:42:29	1.1
+++ ./src/tiff/tif_dir.c	2002/10/20 04:15:26
@@ -119,4 +119,5 @@
 	int status = 1;
 	uint32 v32;
+	uint16 v16;
 	int i, v;
 	double d;
@@ -153,5 +154,5 @@
 		break;
 	case TIFFTAG_COMPRESSION:
-		v = va_arg(ap, int) & 0xffff;
+		v16 = va_arg(ap, int) & 0xffff;
 		/*
 		 * If we're changing the compression scheme,
@@ -160,5 +161,5 @@
 		 */
 		if (TIFFFieldSet(tif, FIELD_COMPRESSION)) {
-			if (td->td_compression == v)
+			if (td->td_compression == v16)
 				break;
 			(*tif->tif_cleanup)(tif);
@@ -168,6 +169,6 @@
 		 * Setup new compression routine state.
 		 */
-		if( (status = TIFFSetCompressionScheme(tif, v)) != 0 )
-			td->td_compression = v;
+		if( (status = TIFFSetCompressionScheme(tif, v16)) != 0 )
+			td->td_compression = v16;
 		break;
 	case TIFFTAG_PHOTOMETRIC:
Index: ./src/tiff/tif_getimage.c
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/tiff/tif_getimage.c,v
retrieving revision 1.2
diff -b -u -2 -r1.2 tif_getimage.c
--- ./src/tiff/tif_getimage.c	2000/02/16 17:12:55	1.2
+++ ./src/tiff/tif_getimage.c	2002/10/20 04:15:27
@@ -1543,10 +1543,10 @@
     TIFFGetFieldDefaulted(img->tif, TIFFTAG_YCBCRSUBSAMPLING, &hs, &vs);
     switch ((hs<<4)|vs) {
-    case 0x44: return (putcontig8bitYCbCr44tile);
-    case 0x42: return (putcontig8bitYCbCr42tile);
-    case 0x41: return (putcontig8bitYCbCr41tile);
-    case 0x22: return (putcontig8bitYCbCr22tile);
-    case 0x21: return (putcontig8bitYCbCr21tile);
-    case 0x11: return (putcontig8bitYCbCr11tile);
+    case 0x44: return (&putcontig8bitYCbCr44tile);
+    case 0x42: return (&putcontig8bitYCbCr42tile);
+    case 0x41: return (&putcontig8bitYCbCr41tile);
+    case 0x22: return (&putcontig8bitYCbCr22tile);
+    case 0x21: return (&putcontig8bitYCbCr21tile);
+    case 0x11: return (&putcontig8bitYCbCr11tile);
     }
     return (NULL);
Index: ./src/tiff/tiff.dsp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/tiff/tiff.dsp,v
retrieving revision 1.5
diff -b -u -2 -r1.5 tiff.dsp
Index: ./src/tiff/tiff.dsw
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/tiff/tiff.dsw,v
retrieving revision 1.3
diff -b -u -2 -r1.3 tiff.dsw
Index: ./src/tiff/tiffiop.h
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/tiff/tiffiop.h,v
retrieving revision 1.3
diff -b -u -2 -r1.3 tiffiop.h
--- ./src/tiff/tiffiop.h	2001/12/15 23:18:18	1.3
+++ ./src/tiff/tiffiop.h	2002/10/20 04:15:28
@@ -214,5 +214,5 @@
 extern	void  TIFFFreeDirectory(TIFF*);
 extern	int  TIFFDefaultDirectory(TIFF*);
-extern	int  TIFFSetCompressionScheme(TIFF*, int);
+extern	int  TIFFSetCompressionScheme(TIFF*, uint16);
 extern	int  TIFFSetDefaultCompressionState(TIFF*);
 extern	uint32 _TIFFDefaultStripSize(TIFF*, uint32);

 	  	 
